tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.hpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <optional>
+#include <vector>
+
+#include "tt_dnn/op_library/operation.hpp"
+#include "tt_eager/tensor/tensor.hpp"
+
+namespace tt {
+namespace operations {
+namespace primary {
+
+using namespace tt_metal;
+
+struct MorehAdam {
+    bool inplace;
+    float lr;
+    float beta1;
+    float beta2;
+    float eps;
+    float weight_decay;
+    uint32_t step;
+    bool amsgrad;
+
+    MemoryConfig output_mem_config;
+
+    void validate(
+        const std::vector<Tensor> &input_tensors,
+        const std::vector<std::optional<const Tensor>> &optional_input_tensors) const;
+    std::vector<Shape> compute_output_shapes(const std::vector<Tensor> &input_tensors) const;
+    std::vector<Tensor> create_output_tensors(const std::vector<Tensor> &input_tensors) const;
+    operation::ProgramWithCallbacks create_program(
+        const std::vector<Tensor> &input_tensors,
+        const std::vector<std::optional<const Tensor>> &optional_input_tensors,
+        std::vector<Tensor> &output_tensors) const;
+    tt::stl::reflection::Attributes attributes() const;
```

Comment:
https://github.com/tenstorrent-metal/tt-metal/commit/53af9512130441d2aeafeec4c87b41bebd93669c#diff-4a6afa7711800ac1d0099139ea3767d10f29491cf05c063b3b9aa315faab058b

This seems like a deprecated style.
TT changed attribute styles to reduce CPU time.

Reply:
applied the feedback